addons/sale_project/models/sale_order.py

```diff
@@ -203,6 +204,18 @@ def action_view_milestone(self):
             }
         }
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        project = self.env['project.project'].browse(self.env.context.get('create_for_project'))
+        created_records = super().create(vals_list)
+        if project:
```

Comment:
```suggestion
    def create(self, vals_list):
        orders = super().create(vals_list)
        project = self.env['project.project'].browse(self.env.context.get('create_for_project_id'))
        if project:
```
nitpicking
* keep the super call first unless there is a specific reason to not keep it first
* `create_for_project` could be renamed somehow to clearly hint that it holds a project id and not a record